### PR TITLE
Add a test case to show #9376 and #9377 have been resolved

### PR DESCRIPTION
**Pull request by mpdude**
This is part of the series of issues fixed by #10547. In particular, the changes from #10566 were relevant.

See #9376 for the bug description.

The suggested fix in #9377 is obsoleted by #10547.

### CODE REVIEW

File: tests/Doctrine/Tests/ORM/Functional/Ticket/GH9376Test.php [hunk 1]
```diff
@@ -0,0 +1,133 @@
...
+    protected function tearDown(): void
+    {
+        $this->_schemaTool->dropSchema([
+            $this->_em->getClassMetadata(GH9376GiftVariant::class),
+            $this->_em->getClassMetadata(GH9376Product::class),
+            $this->_em->getClassMetadata(GH9376Gift::class),
+        ]);
+
+        parent::tearDown();
+    }
```
Comment (greg0ire) [inline tests/Doctrine/Tests/ORM/Functional/Ticket/GH9376Test.php:25]: Please remove `tearDown()` and use `createSchemaForModels()`. See https://github.com/doctrine/orm/pull/9526 for more details about the why.

Comment (mpdude) [reply tests/Doctrine/Tests/ORM/Functional/Ticket/GH9376Test.php:25]: Updated.

File: tests/Doctrine/Tests/ORM/Functional/Ticket/GH9376Test.php [hunk 2]
```diff
@@ -0,0 +1,122 @@
...
+    public function __construct(GH9376Gift $gift)
+    {
+        $this->gift = $gift;
+    }
+}
+
+/**
+ * @Entity
+ */
+
```
Comment (greg0ire) [inline tests/Doctrine/Tests/ORM/Functional/Ticket/GH9376Test.php]:
```suggestion
```

File: tests/Doctrine/Tests/ORM/Functional/Ticket/GH9376Test.php [hunk 3]
```diff
@@ -0,0 +1,121 @@
...
+     * @Column(type="integer")
+     * @GeneratedValue
+     */
+    public $id;
+
+    /**
+     * @ORM\ManyToOne(targetEntity=GH9376Gift::class)
+     * @ORM\JoinColumn(nullable=false)
+     *
+     * @var GH9376Gift
```
Comment (greg0ire) [inline tests/Doctrine/Tests/ORM/Functional/Ticket/GH9376Test.php:74]: The variant points to the gift and nothing else

File: tests/Doctrine/Tests/ORM/Functional/Ticket/GH9376Test.php [hunk 4]
```diff
@@ -0,0 +1,121 @@
...
+    public function __construct(GH9376Gift $gift)
+    {
+        $this->gift = $gift;
+    }
+}
+
+/**
+ * @Entity
+ */
+class GH9376Product
```
Comment (greg0ire) [inline tests/Doctrine/Tests/ORM/Functional/Ticket/GH9376Test.php:87]: The product points to nothing

File: tests/Doctrine/Tests/ORM/Functional/Ticket/GH9376Test.php [hunk 5]
```diff
@@ -0,0 +1,121 @@
...
+     * @Id @Column(type="integer")
+     * @GeneratedValue
+     */
+    public $id;
+}
+
+/**
+ * @Entity
+ */
+class GH9376Gift
```
Comment (greg0ire) [inline tests/Doctrine/Tests/ORM/Functional/Ticket/GH9376Test.php:100]: The gif points to a product and nothing else.

So the graph is like Variant -> Gift -> Product

:thinking: where is the circular reference?

Comment (mpdude) [reply tests/Doctrine/Tests/ORM/Functional/Ticket/GH9376Test.php:100]: 👀

Comment (mpdude) [reply tests/Doctrine/Tests/ORM/Functional/Ticket/GH9376Test.php:100]: What a careful reader you are. Test originally comes from here: https://github.com/doctrine/orm/pull/9377/files#diff-aa92ab1b41151bbdb1e6af1f38699ad708680e99bdf4420b56b9eddd64376061

**Comment by mpdude** [comment]
I am closing this for the time being, since the test provided in #9377 does not fail on 2.16 and the bug description in #9376 is insufficient to reproduce the problem.

**Pull request closed without merging**